lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2017, Nathan Davison <ndavison85@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: listen_ports_facts
+
+author:
+    - Nathan Davison (@ndavison)
+
+version_added: "2.8"
+
+description:
+    - Gather facts on processes listening on TCP and UDP ports.
+    - Optionally provide whitelists to gather facts on violations of the whitelist.
+
+short_description: Gather facts on processes listening on TCP and UDP ports.
+
+options:
+  whitelist_tcp:
+    description:
+      - A list of TCP port numbers that are expected to have processes listening.
+    type: list
+    default: []
+  whitelist_udp:
+    description:
+      - A list of UDP port numbers that are expected to have processes listening.
+    type: list
+    default: []
+'''
+
+EXAMPLES = r'''
+- name: Gather facts on listening ports, and populate the tcp_listen_violations fact with processes listening on TCP ports that are not 22, 80, or 443.
+  listen_ports_facts:
+    whitelist_tcp:
+      - 22
+      - 80
+      - 443
+
+- name: TCP whitelist violation
+  debug:
+    msg: TCP port {{ item.port }} by pid {{ item.pid }} violates the whitelist
+  with_items: "{{ tcp_listen_violations }}"
+  when: tcp_listen_violations
```

Comment:
Add a few more examples of just listing the ports:
```suggestion
  when: tcp_listen_violations
  - name: List TCP ports
    debug:
      var: ansible_facts.tcp_listen  | map(attribute='port') | sort | list

  - name: List UDP ports
    debug:
      var: ansible_facts.udp_listen | map(attribute='port') | sort | list

  - name: List all ports
    debug:
      var: (ansible_facts.tcp_listen + ansible_facts.udp_listen) | map(attribute='port') | unique | sort | list
```